platform/ios/src/MGLMapView.mm

```diff
@@ -389,8 +389,14 @@ - (nonnull NSURL *)styleURL
     }
 
     NSString *styleURLString = @(self.mbglMap.getStyle().getURL().c_str()).mgl_stringOrNilIfEmpty;
-    MGLAssert(styleURLString, @"Invalid style URL string %@", styleURLString);
-    return styleURLString ? [NSURL URLWithString:styleURLString] : nil;
+
+    if (!styleURLString)
+    {
+        MGLLogInfo(@"Warning: Invalid style URL string %@", styleURLString);
+    }
+    
+    return styleURLString ? [NSURL URLWithString:styleURLString] :
+        [MGLStyle streetsStyleURLWithVersion:MGLStyleDefaultVersion];
```

Comment:
@friedbunny how do you feel about changing the return value here? Although it would be more correct it *is* a change in behavior.

Reply:
So long as we changelog it, I think this is justifiably a “bug fix”.

Previously, did a `nil` return value here actually have some special meaning elsewhere? Just above this is `- (nonnull NSURL *)styleURL {` and [publicly](https://github.com/mapbox/mapbox-gl-native/blob/3b1d437460939254191479bd730599c2418c20aa/platform/ios/src/MGLMapView.h#L274) it’s marked `null_resettable`. We document that setting this to `nil` will “use the default style”, so I’m not sure if changing the default return value here is necessary — is it already happening elsewhere?

Reply:
Before this change, the return value shouldn’t have been reachable code, technically. Setting `styleURL` to `nil` would cause it to get filled in with the default style:

https://github.com/mapbox/mapbox-gl-native/blob/869ff70e0dbfeab16b20bb1a8d765db153312a42/platform/ios/src/MGLMapView.mm#L398-L401

So getting `styleURL` after that point would come up with the default value. The assertion is meant to ensure that’s the case (but honestly is just there to satisfy a static analysis warning).

A `nil` style URL coming out of mbgl would be an internal error. The rule of thumb is that assertions are for internal errors out of the developer’s control, whereas an exception (or possibly a log message) are for developer error or environmental issues like network connection failures.

> Just above this is `- (nonnull NSURL *)styleURL {` and [publicly](https://github.com/mapbox/mapbox-gl-native/blob/3b1d437460939254191479bd730599c2418c20aa/platform/ios/src/MGLMapView.h#L274) it’s marked `null_resettable`.

For a `null_resettable` property, autosynthesis would come up with a getter with a non-null return value and a setter with a nullable value parameter.

Reply:
> A `nil` style URL coming out of mbgl would be an internal error. The rule of thumb is that assertions are for internal errors out of the developer’s control, whereas an exception (or possibly a log message) are for developer error or environmental issues like network connection failures.

> For a `null_resettable` property, autosynthesis would come up with a getter with a non-null return value and a setter with a nullable value parameter.

Gotcha, this all makes sense. Thanks @1ec5.

> The assertion is meant to ensure that’s the case

Are there other assertions in the code base (after where a `nil` styleURL becomes the default value) that fulfill this same function? 

>  (but honestly is just there to satisfy a static analysis warning).

I'm not sure what "satisfying a static analysis warning" means. Is it important? Hypothetically speaking, what if we were to get rid of this assertion altogether?